src/OrbitQt/orbitmainwindow.cpp

```diff
@@ -782,10 +782,14 @@ void OrbitMainWindow::OnTimer() {
   app_->MainTick();
 
   for (OrbitGLWidget* gl_widget : gl_widgets_) {
-    gl_widget->update();
+    if (gl_widget->GetCanvas() != nullptr && gl_widget->GetCanvas()->GetNeedsRedraw()) {
```

Comment:
Probably a stupid question, but are you sure about this? What happens if GetNeedsRedraw was set into false after a redrawing the CaptureWindow?

Reply:
I'm not sure I understand the issue you're seeing. "Needs redraw" is set whenever anything happened that requires redraw, and is only cleared when that redraw happened. If it returns `false` at this point, it should be save to assume all contents of the capture window are drawn and up-to-date, or is there some case I'm missing?

Reply:
Yes, of course. Was a stupid question :P